presto-main/src/test/java/com/facebook/presto/security/TestFileBasedSystemAccessControl.java

```diff
@@ -57,37 +56,71 @@ public void testCatalogRules()
 
         transaction(transactionManager, accessControlManager)
                 .execute(transactionId -> {
+                    assertEquals(accessControlManager.filterCatalogs(admin, allCatalogs), allCatalogs);
                     Set<String> aliceCatalogs = ImmutableSet.of("open-to-all", "alice-catalog", "all-allowed");
                     assertEquals(accessControlManager.filterCatalogs(alice, allCatalogs), aliceCatalogs);
+                    Set<String> bobCatalogs = ImmutableSet.of("open-to-all", "all-allowed");
+                    assertEquals(accessControlManager.filterCatalogs(bob, allCatalogs), bobCatalogs);
+                    Set<String> nonAsciiUserCatalogs = ImmutableSet.of("open-to-all", "all-allowed", "ÅÅÅ");
+                    assertEquals(accessControlManager.filterCatalogs(nonAsciiUser, nonAsciiUserCatalogs), nonAsciiUserCatalogs);
+                });
+    }
+
+    @Test
+    public void testSchemaOperations()
+    {
+        TransactionManager transactionManager = createTestTransactionManager();
+        AccessControlManager accessControlManager = new AccessControlManager(transactionManager);
+
+        accessControlManager.setSystemAccessControl(FileBasedSystemAccessControl.NAME, ImmutableMap.of(
+                "security.config-file", "src/test/resources/catalog.json"));
+
+        transaction(transactionManager, accessControlManager)
+                .execute(transactionId -> {
                     Set<String> aliceSchemas = ImmutableSet.of("schema");
                     assertEquals(accessControlManager.filterSchemas(transactionId, alice, "alice-catalog", aliceSchemas), aliceSchemas);
+                    assertEquals(accessControlManager.filterSchemas(transactionId, bob, "alice-catalog", aliceSchemas), ImmutableSet.of());
+
+                    accessControlManager.checkCanCreateSchema(transactionId, alice, aliceSchema);
```

Comment:
Such situation won't arise. In order for the test to pass, all the checks that are supposed to pass are outside of try() block. The check that throws is inside the try() block, so if it passes incorrectly, the test will fail.